lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -129,14 +129,37 @@ def generate_refresh_token
     end
 
     def generate_token
+      set_timestamps
+
       generator = Doorkeeper.configuration.access_token_generator.constantize
       self.token = generator.generate(resource_owner_id: resource_owner_id,
                                       scopes: scopes, application: application,
-                                      expires_in: expires_in)
+                                      expires_in: expires_in, created_at: created_at)
     rescue NoMethodError
       raise Errors::UnableToGenerateToken, "#{generator} does not respond to `.generate`."
     rescue NameError
       raise Errors::TokenGeneratorNotFound, "#{generator} not found"
     end
+
+    def set_timestamps
+      if self.created_at.nil?
+        self.created_at = current_time
+
+        if respond_to? :updated_at=
+          # Mongoid will not set the updated_at automatically if the created_at is set manually
+          self.updated_at = current_time
+        end
+      end
+    end
+
+    def current_time
+      if self.class.respond_to? :default_timezone
+        # Duplicate behavior from Rails 4
+        self.class.default_timezone == :utc ? Time.now.utc : Time.now
```

Comment:
I see it comes from Rails, and I still feel rather uncomfortable because I don't understand it. In the case `self.class.default_timezone` is actually `utc`, wouldn't `Time.now.utc == Time.now`? And, what would happen if `:utc` were a string instead.

I don't want to change behavior to client Rails applications, but I feel so compelled to doing always `Time.now.utc` at the same time...